app/controllers/heritages_controller.rb

```diff
@@ -22,7 +23,7 @@ def create
   end
 
   def update
-    @heritage = BuildHeritage.new(permitted_params).execute
+    @heritage = BuildHeritage.new(permitted_params, district: @district).execute
```

Comment:
Thanks this looks like a very clean way to handle it. Its also a very implicit `if` statement, so I think we should have a test to get the heritage for the following cases:

1. endpoint has no district, name passed in
2. endpoint has no district, name and district passed in
3. endpoint has district, name passed in
4. endpoint has district, name and district passed in
5. endpoint has district, with another endpoint in different district with the same name, name passed in
6. endpoint has district, with another endpoint in different district with the same name, name and district passed in

Reply:
@davidsiaw just to confirm, what do you mean by `endpoint`here?
like `endpoint has no district, name passed in`
this is the case when pass heritage name but there is no district?

Reply:
Not pass, but the database has an endpoint but district is set to `nil`

Reply:
district has endpionts right?
https://github.com/degica/barcelona/blob/07009e6736178a39b36816517af55a766db585c8/app/models/district.rb#L12-L12

Reply:
also district name should be unique
also I could not understand what you mean by 🙏 
```
endpoint has district, with another endpoint in different district with the same name, name passed in
``

Reply:
@davidsiaw updated the logic
https://github.com/degica/barcelona/pull/667/commits/05f4bae33902f14216b35fc29639dbad47ffb87e
also existing tests cover for when no district name is given

Reply:
@Yuuki77 I think it is hard to write this test from what I see, because you are testing the API. You should test BuildHeritage instead, and it will be clearer. What we need to test essentially is all the permutations of

1. The various states of the database
2. The inputs passed to update

Reply:
@davidsiaw added the test.
I think the other scenario is that the wrong district is passed (which has always been possible).
So I added the checks.
other that existing tests should cover.
The change I made here was to pass `district` in update, which was nil.
 https://github.com/degica/barcelona/pull/667/commits/dd9fef7afc8f901c0d79de894c7b1ec18cbee9a8